tests/integration/defs/perf/utils.py

```
@@ -347,6 +350,232 @@ def get_cmd_str(self, cmd_idx) -> List[str]:
+
+        # Generate server config
+        server_config = {
+            'hostname': current_hostname,
+            'port': server_port,
+            'backend': 'pytorch',
+            'context_servers': {
+                'num_instances': num_ctx_servers,
+                'urls': [f'{host}:{worker_port}' for host in ctx_hostnames]
+            },
+            'generation_servers': {
+                'num_instances': num_gen_servers,
```

Ruff (0.14.4)

Undefined name `num_gen_servers`

(F821)